src/VisualStudio/Core/Impl/CodeModel/InternalElements/AbstractCodeElement.cs

```diff
@@ -119,6 +120,9 @@ protected virtual string GetFullName()
             return CodeModelService.GetFullName(node, semanticModel);
         }
 
+        protected DocumentOptionSet GetDocumentOptions(CancellationToken cancellationToken)
+            => State.ThreadingContext.JoinableTaskFactory.Run(() => GetDocument().GetOptionsAsync(cancellationToken));
```

Comment:
📝 Synchronous invocation of an asynchronous method is a code smell which we don't want to hide through encapsulation. This method should be inlined to each caller, even if that means the caller looks "icky".